### ISSUE TITLE
source command does not have an escape sequence for the markdown

### ISSUE THREAD

**Issue by bhagyajbijukumar** [Gorialis/jishaku#79]
![image](https://user-images.githubusercontent.com/62506150/100105224-1dca5d80-2e8d-11eb-9b08-df00a0d18285.png)
see the above image

**Comment by iomintz** [comment]
I started implementing this by just making the `jsk_source` command replace all occurrences of <code>```</code> with <code>``\N{zero width space}`</code>, but then I noticed that since FilePaginator also uses WrappedPaginator with a fenced code block, it might be better to place this escaping code inside a new `commands.Paginator` subclass that's used as the base for all other base Paginator classes. Something like this:

```py
def __init__(self, ...):
    self.escape_code_blocks = False
    if '```' in prefix and '```' in suffix:
        self.escape_code_blocks = True

def add_line(self, line):
    if self.escape_code_blocks:
        line = line.replace('```', '``\N{zero width space}`')
        super().add_line(line)
```

Let me know what you think @Gorialis

**Comment by bhagyajbijukumar** [comment]
i have earlier used discord.utils.escape_markdown from the discordpy library in my project isn't it possible to use that function?

**Comment by Gorialis** [comment]
I merged #92 a while ago, but I realized in auditing later that it actually produces a lot of formatting inconsistencies (all underscores get "escaped"). I'll probably end up favoring [this implementation](https://github.com/Gorialis/jishaku/issues/79#issuecomment-733592980), but we'll see how it goes when I have time to do some proper QA.

**Comment by Gorialis** [comment]
Discord recently added an interactive, syntax highlighted, file preview feature on the desktop and web clients.

As this is the most common usage vector for Jishaku, and the intuitiveness greatly exceeds home-rolled solutions, as of 9099fe0cd1a3190b894ec4083d2ca661696ae279, Jishaku will now check the file upload limit and, if the source file content fits within it, will upload the content as a file instead.

![image](https://user-images.githubusercontent.com/8049998/113153118-bf195400-9271-11eb-8758-cc64bb55aff9.png)

This circumvents the need for escaping in this case, but in the case it does not fit, the handling has been changed to favor @iomintz 's implementation.
